HW_2_updated.py

```diff
@@ -0,0 +1,45 @@
+import random
+import string
+
+def get_the_list():
+    uni_list = []
+    # get the number of key/values in a dict + dicts quantity
+    quantity = random.randint(2,10)
+    for i in range(quantity):
+        # generate lists for keys and values
+        keys = random.sample(string.ascii_lowercase, quantity)
+        values = random.sample(range (0, 100), quantity)
+        # combine them into dict
+        my_dict = dict(zip(keys, values))
+        # append new dict to a list
+        uni_list.append(my_dict)
+    print(uni_list)
+    return uni_list
+
+
+def create_common_dict(uni_list):
+    common_dict = {'0': 0}
+    i = 0
+    for dictionary in uni_list:
+        for key, value in dictionary.items():
+            firsts = [w[0] for w in list(common_dict)]
+            if key not in firsts:
+                common_dict.update({key: value})
+            else:
+                for k in list(common_dict.keys()):
+                    if k[0] == key:
+                        if dictionary.get(key) > common_dict.get(k):
+                            common_dict.update({key + '_' + str(i): value})
+                            del common_dict[k]
+                        else:
+                            common_dict.update({key + '_' + str([i for i, d in enumerate(uni_list) if d.get(key) == common_dict.get(k)]): common_dict.get(k)})
+                            del common_dict[k]
+        i += 1
+    del common_dict['0']
+    print(common_dict)
+    return common_dict
+
+
+if __name__=='__main__':
```

Comment:
Your function **create_common_dict** works well for some values, but also it deletes a lot of keys.

Tested your code on the following dataset:
```
rand_lst = [{'m': 84, 'f': 41, 'l': 34, 'c': 33, 'q': 58, 'h': 14, 'p': 25, 't': 85, 'x': 70, 'g': 3, 'r': 63, 'y': 93, 'd': 44, 'o': 79, 'n': 73},
{'n': 86, 'l': 1, 'w': 68, 'h': 52, 'e': 91, 'x': 31, 'f': 58, 'b': 45, 'a': 45, 'q': 100, 'r': 39, 'p': 88, 'z': 27, 'j': 66},
{'p': 35, 'd': 73, 'j': 75, 'v': 89, 'm': 81, 'l': 33, 'z': 60, 'h': 59, 'u': 82, 's': 96, 'i': 86, 'b': 68, 'w': 100, 'x': 38, 'a': 50, 'o': 67},
{'p': 26, 'f': 74, 'o': 63, 'v': 4, 'e': 2, 'r': 39, 'k': 34, 'w': 95, 'g': 51, 'n': 47, 'h': 27, 'c': 29, 'z': 0, 'q': 33, 'a': 69, 't': 87}]
```

Expected | Actual
-----------| --------
m_1': 84   | m_[0]': 84
'h_3': 59   | 'h_[2]': 59
'l_1':34     | not exist
'p_2': 88  | not exist
'x_1': 70 | not exist
'r_1': 63 | not exist
'o_1': 79 | not exist








If you are interested in my solution:
```
from random import randint, choice
from string import ascii_lowercase


rand_list = [{choice(ascii_lowercase): randint(0, 100) for i in range(len(ascii_lowercase))} for j in range(randint(2, 10))]
print(rand_list)
final_dict, tmp_dict = {},  {}

#Transform from list of dicts into dict of lists.
for dictionary in rand_list:
    for k, v in dictionary.items():
        tmp_dict.setdefault(k, []).append(v)

#Now choose only the biggest one
for k, v in tmp_dict.items():
    if len(v) > 1:
        final_dict[k+"_"+str(v.index(max(v))+1)] = max(v)
    else:
        final_dict[k] = v[0]
print(final_dict)
```

Reply:
It's perfect! Simple and effective!
Hope some day I'll be able to write this way!